src/test/java/com/team4/testingsystem/controllers/CoachGradeControllerIntegrationTest.java

```diff
@@ -0,0 +1,253 @@
+        List<CoachGradeDTO> grades = objectMapper.readValue(response, new TypeReference<>() {});
+
+        Assertions.assertEquals(List.of(new CoachGradeDTO(grade)), grades);
+    }
+
+    @Test
+    void getGradesTwoGrades() throws Exception {
+        com.team4.testingsystem.entities.Test test = EntityCreatorUtil.createTest(user);
+        testsRepository.save(test);
+
+        Question question = EntityCreatorUtil.createQuestion(user);
+        questionRepository.save(question);
+
+        CoachGrade grade = new CoachGrade(test, question, 8);
+        gradeRepository.save(grade);
+
+        Question question2 = EntityCreatorUtil.createQuestion(user);
+        questionRepository.save(question2);
+
+        CoachGrade grade2 = new CoachGrade(test, question, 7);
+        gradeRepository.save(grade2);
+
+        MvcResult mvcResult = mockMvc.perform(get("/grades/{testId}", test.getId())
+                .with(user(userDetails)))
+                .andExpect(status().isOk())
+                .andReturn();
+
+        String response = mvcResult.getResponse().getContentAsString();
+        List<CoachGradeDTO> grades = objectMapper.readValue(response, new TypeReference<>() {});
+
+        Assertions.assertEquals(2, grades.size());
+        Assertions.assertTrue(grades.contains(new CoachGradeDTO(grade)));
+        Assertions.assertTrue(grades.contains(new CoachGradeDTO(grade2)));
+    }
+
+    @Test
+    void createGrade() throws Exception {
+        CoachGradeDTO gradeDTO = CoachGradeDTO.builder()
+                .testId(200L)
+                .questionId(300L)
+                .grade(10)
+                .build();
+
+        mockMvc.perform(post("/grades/")
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(objectMapper.writeValueAsString(gradeDTO))
+                .with(user(userDetails)))
+                .andExpect(status().isNotFound());
+
+        com.team4.testingsystem.entities.Test test = EntityCreatorUtil.createTest(user);
+        testsRepository.save(test);
+
+        gradeDTO.setTestId(test.getId());
+
+        mockMvc.perform(post("/grades/")
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(objectMapper.writeValueAsString(gradeDTO))
+                .with(user(userDetails)))
+                .andExpect(status().isNotFound());
+
+        Question question = EntityCreatorUtil.createQuestion(user);
+        questionRepository.save(question);
+
+        gradeDTO.setQuestionId(question.getId());
+
+        mockMvc.perform(post("/grades/")
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(objectMapper.writeValueAsString(gradeDTO))
+                .with(user(userDetails)))
+                .andExpect(status().isOk());
+
+        Optional<CoachGrade> grade = gradeRepository.findByTestAndQuestion(test, question);
+        Assertions.assertTrue(grade.isPresent());
+        Assertions.assertEquals(gradeDTO.getGrade(), grade.get().getGrade());
```

Comment:
What do you expect from the test? Are you trying to get a grade through mockMvc and from the controller, and comparing the results?

Let's try to compare mockMvc result with a grade we saved before

Reply:
`gradeDTO` is created at the beginning of the test, with hard-coded values, and is send in `create` request. `grade` is the value from the DB, that was saved during the request.
It's `createGrade` test, so I'm checking that the grade, saved by the controller to the DB, is equal to the value `10` from the `gradeDTO`, that was in the request.